src/util/psbt/serialize.rs

```diff
@@ -85,3 +87,30 @@ impl Deserialize for (Fingerprint, Vec<ChildNumber>) {
         Ok((fprint, dpath))
     }
 }
+
+// partial sigs
+impl Serialize for Vec<u8> {
+    fn serialize(&self) -> Vec<u8> {
+        self.clone()
+    }
+}
+
+impl Deserialize for Vec<u8> {
+    fn deserialize(bytes: &Vec<u8>) -> Result<Self, encode::Error> {
+        Ok(bytes.clone())
+    }
+}
+
+impl Serialize for SigHashType {
+    fn serialize(&self) -> Vec<u8> {
+        serialize(&self.as_u32())
+    }
+}
+
+impl Deserialize for SigHashType {
+    fn deserialize(bytes: &Vec<u8>) -> Result<Self, encode::Error> {
+        let ref mut d = Cursor::new(&bytes[..]);
+
+        Ok(SigHashType::from_u32(Decodable::consensus_decode(d)?))
```

Comment:
You need to check for trailing bytes here; I also think we should check that the resulting sighashtype roundtrips, i.e. `.as_u32()` gives the same thing as went into `from_u32`.